src/content_cao.cpp

```diff
@@ -1786,6 +1831,14 @@ void GenericCAO::processMessage(const std::string &data)
 	}
 }
 
+void GenericCAO::knockback(const v3f &direction, f32 time_knockback)
+{
+	m_knockback_timer = time_knockback;
```

Comment:
We don't have stun after any other damage or after combat damage currently, it would be very frustrating to introduce it now and be paralysed on every enemy hit, it would ruin combat.
It is an intended feature of MT that the player can move while in mid-air, although that is unrealistic, there was a suggestion to remove mid-air movement a while ago and it was strongly refused.
Not having a stun timer that disables controls also makes this simpler, all this needs is an instantaneous setting of player velocity.